liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,500 @@
+      /* eslint-enable no-param-reassign */
+    }));
+
+    // // Initialize token objects
+    ts('Checking for new tokens');
+    await Promise.all(Array.from(tokenSet).map(async (token) => {
+      await verifyToken(data, token);
+    }));
+
+    // Grab token balances from on-chain
+    ts('Updating balances');
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      await Promise.all(data.accounts[currentAccount].assetsIn.map(async (currentToken) => {
+        const snapshot = await data.tokens[currentToken].contract
+          .getAccountSnapshot(currentAccount);
+
+        /* eslint-disable no-param-reassign */
+        if (snapshot && snapshot[1].toString() !== '0') {
+          data.supply[currentToken][currentAccount] = BigNumber(snapshot[1].toString())
+            .dividedBy(data.tokens[currentToken].cTokenDecimalsMult);
+          const tokenQty = data.supply[currentToken][currentAccount]
+            .multipliedBy(data.tokens[currentToken].exchangeRateMult);
+          ts(
+            `User ${currentAccount
+            } supplied ${Math.round(tokenQty)
+            } ${data.tokens[currentToken].symbol
+            } ( ${Math.round(data.supply[currentToken][currentAccount])
+            } cTokens ) `,
+          );
+        }
+        if (snapshot && snapshot[2].toString() !== '0') {
+          data.borrow[currentToken][currentAccount] = BigNumber(snapshot[2].toString())
+            .dividedBy(data.tokens[currentToken].tokenDecimalsMult);
+          ts(
+            `User ${currentAccount
+            } borrowed ${Math.round(data.borrow[currentToken][currentAccount])
+            } ${data.tokens[currentToken].symbol}`,
+          );
+        }
+        /* eslint-enable no-param-reassign */
+        return null;
+      }));
+    }));
+    // #endregion
+
+    // #region Update all token prices via 1inch for now.
+    ts('Updating token prices and collateral factors');
+    // To-Do: Switch the primary feed back to ChainLink
+    await Promise.all(Object.keys(data.tokens).map(async (currentToken) => {
+      const price = await oneInchContract.getRateToEth(data.tokens[currentToken].underlying, 0);
+      // Adjust for native decimals
+      const oneInchMult = BigNumber(10).pow(36 - data.tokens[currentToken].tokenDecimals);
+      /* eslint-disable no-param-reassign */
+      data.tokens[currentToken].price = BigNumber(price.toString())
+        .dividedBy(oneInchMult);
+      // Update the Collateral Factor
+      const market = await comptrollerContract.markets(currentToken);
+      data.tokens[currentToken].collateralMult = BigNumber(market[1].toString())
+        .dividedBy(BigNumber(10).pow(18));
+      /* eslint-enable no-param-reassign */
+      ts(
+        `Updated price of ${data.tokens[currentToken].symbol
+        } is ${data.tokens[currentToken].price.toString()
+        } ETH with a collateral factor of ${data.tokens[currentToken].collateralMult}`,
+      );
+    }));
```

Comment:
You should be able to use `entries` in place of `keys` to make the code more readable.
```suggestion
    await Promise.all(Object.entries(data.tokens).map(async (currentToken, tokenData) => {
      const price = await oneInchContract.getRateToEth(tokenData.underlying, 0);
      
      // Adjust for native decimals
      const oneInchMult = BigNumber(10).pow(36 - tokenData.tokenDecimals);
      tokenData.price = BigNumber(price.toString()).dividedBy(oneInchMult);
      
      // Update the Collateral Factor
      const market = await comptrollerContract.markets(currentToken);
      tokenData.collateralMult = BigNumber(market[1].toString())
        .dividedBy(BigNumber(10).pow(18));
    }));
```